streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -40,6 +41,9 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        if (streamThread.state() == StreamThread.State.DEAD) {
```

Comment:
I see your point, but I am not sure if this would make the code too convoluted. Basically, we can only query in state `RUNNING`, want to ignore in state `DEAD` and want to fail for the other 3 states. Right now, the code is quite explicit about this.

If you insist, we could got with `StreamThread#isQueryable()` but we would need a comment why a DEAD thread is queryable. Thoughts?